### DISCUSSION TITLE
Public the _inputXml variable from WebserviceRequest

### DISCUSSION THREAD

**Discussion by sefirosweb** [PrestaShop/PrestaShop#31738] [Ideas]
### The problem

Hello, I'm writing modile with a custom webservice api, this is extended from WebserviceSpecificManagementInterface

When i'm trying to get data send from api i found I can't get this data because is protected, and no have a get function for this value,

https://github.com/PrestaShop/PrestaShop/blob/e4704c25c1c1a94127eaddcc71185f639f50ca8a/classes/webservice/WebserviceRequest.php#L166

This have sense?

### The solution

Change: https://github.com/PrestaShop/PrestaShop/blob/e4704c25c1c1a94127eaddcc71185f639f50ca8a/classes/webservice/WebserviceRequest.php#L166

to public

or add getInputXml function:
```php
    public function getInputXml()
    {
        return $this->_inputXml;
    }
```

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by FabienPapet** [comment]
I'm not a big fan of public properties for various reasons including maintainability. But adding a getter would be a nice improvement IMO.
